src/pages/ThorChainLP/components/AddLiquitity/AddLiquidityInput.tsx

```diff
@@ -118,28 +122,94 @@ export const AddLiquidityInput: React.FC<AddLiquidityInputProps> = ({
   const translate = useTranslate()
   const { history: browserHistory } = useBrowserRouter()
   const history = useHistory()
-  const divider = useMemo(() => <StackDivider borderColor='border.base' />, [])
 
   const votingPower = useAppSelector(selectVotingPower)
   const isSnapshotApiQueriesPending = useAppSelector(selectIsSnapshotApiQueriesPending)
+  const isSnapInstalled = useIsSnapInstalled()
   const isVotingPowerLoading = useMemo(
     () => isSnapshotApiQueriesPending && votingPower === undefined,
     [isSnapshotApiQueriesPending, votingPower],
   )
 
-  const { data: parsedPools } = usePools()
+  const [poolAsset, setPoolAsset] = useState<Asset | undefined>()
+  const [slippageRune, setSlippageRune] = useState<string | undefined>()
+  const [isSlippageLoading, setIsSlippageLoading] = useState(false)
+  const [shareOfPoolDecimalPercent, setShareOfPoolDecimalPercent] = useState<string | undefined>()
+  const [activeOpportunityId, setActiveOpportunityId] = useState<string | undefined>()
+  const [approvalTxId, setApprovalTxId] = useState<string | null>(null)
+  const [poolAssetAccountAddress, setPoolAssetAccountAddress] = useState<string | undefined>(
+    undefined,
+  )
+
+  // Virtual as in, these are the amounts if depositing symetrically. But a user may deposit asymetrically, so these are not the *actual* amounts
+  // Keeping these as virtual amounts is useful from a UI perspective, as it allows rebalancing to automagically work when switching from sym. type,
+  // while using the *actual* amounts whenever we do things like checking for asset balance
+  const [virtualAssetCryptoLiquidityAmount, setVirtualAssetCryptoLiquidityAmount] = useState<
+    string | undefined
+  >()
+  const [virtualAssetFiatLiquidityAmount, setVirtualAssetFiatLiquidityAmount] = useState<
+    string | undefined
+  >()
+  const [virtualRuneCryptoLiquidityAmount, setVirtualRuneCryptoLiquidityAmount] = useState<
+    string | undefined
+  >()
+  const [virtualRuneFiatLiquidityAmount, setVirtualRuneFiatLiquidityAmount] = useState<
+    string | undefined
+  >()
 
+  const { data: pools } = usePools()
   const assets = useAppSelector(selectAssets)
-  const poolAssets = useMemo(() => {
-    if (!parsedPools) return []
 
-    return [...new Set(parsedPools.map(pool => assets[pool.assetId]).filter(isSome))]
-  }, [assets, parsedPools])
+  const poolAssets = useMemo(() => {
+    return [...new Set((pools ?? []).map(pool => assets[pool.assetId]).filter(isSome))]
+  }, [assets, pools])
+
+  const accounts = useAppSelector(selectPortfolioAccounts)
+  const accountsByAssetId = useMemo(() => {
+    return Object.entries(accounts).reduce<Record<AssetId, AccountId[]>>(
+      (acc, [accountId, account]) => {
+        account.assetIds.forEach(assetId => {
+          if (!acc[assetId]) acc[assetId] = []
+          acc[assetId].push(accountId)
+        })
+        return acc
+      },
+      {},
+    )
+  }, [accounts])
```

Comment:
Feels like this should really be a selector if it doesn't exist already?

Reply:
agreed, thought you might know if it exists somewhere. I couldn't find it though. will move out

Reply:
https://github.com/shapeshift/web/pull/6278/commits/a7f00f10c2f72e4e2fb151a2e0f99f0e632324dd